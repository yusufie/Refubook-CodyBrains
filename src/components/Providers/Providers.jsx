import { useTranslation } from 'react-i18next';
import { FaFacebookF, FaGoogle } from 'react-icons/fa';
import useFacebookAuth from '../../hooks/useFacebookAuth';
import useGoogleAuth from '../../hooks/useGoogleAuth';

const Providers = () => {
  const { t } = useTranslation();

  const { handleLoginWithGoogle } = useGoogleAuth();
  const { handleLoginWithFacebook } = useFacebookAuth();

  return (
    <div className="flex flex-col md:flex-row  items-center justify-center gap-5   sm:px-6 lg:px-8 w-full ">
      <button
        type="button"
        data-testid="google"
        aria-label="google"
        className="bg-googleRed py-4 px-20 w-full inline-flex items-center justify-center border border-gray-300 shadow-sm text-sm font-medium rounded-md text-white  "
        onClick={handleLoginWithGoogle}
      >
        <FaGoogle className="mr-2 text-refubookWhite text-2xl" />
      </button>
      <h3 className="text-center text-lg font-medium text-refubookBlue">
        {t('OR')}
      </h3>
      <button
        type="button"
        data-testid="facebook"
        aria-label="facebook"
        className="bg-facebookBlue py-4 px-20 w-full inline-flex items-center justify-center border border-gray-300 shadow-sm text-sm font-medium rounded-md text-white  "
        onClick={handleLoginWithFacebook}
      >
        <FaFacebookF className="mr-2 text-refubookWhite text-2xl" />
      </button>
    </div>
  );
};

export default Providers;
