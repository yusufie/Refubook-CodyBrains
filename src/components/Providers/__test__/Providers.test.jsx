import renderer from 'react-test-renderer';
import { screen, render, fireEvent } from '@testing-library/react';
import Providers from '../Providers';

jest.mock('react-icons/fa', () => ({
  ...jest.requireActual('react-icons/fa'),
  FaFacebookF: () => jest.fn(),
  FaGoogle: () => jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => jest.fn(),
}));

jest.mock('firebase/auth', () => ({
  ...jest.requireActual('firebase/auth'),
  signInWithPopup: () => jest.fn(),
}));

jest.mock('sweetalert2', () => ({
  ...jest.requireActual('sweetalert2'),
  fire: () => jest.fn(),
}));

jest.mock('../../../services/firebase.config', () => ({
  ...jest.requireActual('../../../services/firebase.config'),
  facebookProvider: () => jest.fn(),
  googleProvider: () => jest.fn(),
}));

jest.mock('../../../features/userSlice', () => ({
  ...jest.requireActual('../../../features/userSlice'),
  login: () => jest.fn(),
}));

jest.mock('react-icons/fa', () => ({
  ...jest.requireActual('react-icons/fa'),
  FaFacebookF: () => <div>FaFacebookF</div>,
  FaGoogle: () => <div>FaGoogle</div>,
}));

describe('Providers', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Providers />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    render(<Providers />);
    const google = screen.getByTestId('google');
    const facebook = screen.getByTestId('facebook');
    fireEvent.click(google);
    fireEvent.click(facebook);
  });
});
