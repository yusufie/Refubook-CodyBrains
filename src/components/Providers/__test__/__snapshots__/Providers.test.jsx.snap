// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Providers should render correctly 1`] = `
<div
  className="flex flex-col md:flex-row  items-center justify-center gap-5   sm:px-6 lg:px-8 w-full "
>
  <button
    aria-label="google"
    className="bg-googleRed py-4 px-20 w-full inline-flex items-center justify-center border border-gray-300 shadow-sm text-sm font-medium rounded-md text-white  "
    data-testid="google"
    onClick={[Function]}
    type="button"
  >
    <div>
      FaGoogle
    </div>
  </button>
  <h3
    className="text-center text-lg font-medium text-refubookBlue"
  >
    OR
  </h3>
  <button
    aria-label="facebook"
    className="bg-facebookBlue py-4 px-20 w-full inline-flex items-center justify-center border border-gray-300 shadow-sm text-sm font-medium rounded-md text-white  "
    data-testid="facebook"
    onClick={[Function]}
    type="button"
  >
    <div>
      FaFacebookF
    </div>
  </button>
</div>
`;
